swift/ql/lib/codeql/swift/dataflow/internal/DataFlowPrivate.qll

```diff
@@ -79,6 +98,8 @@ private class KeyPathComponentPostUpdateNode extends TKeyPathComponentPostUpdate
   }
 
   KeyPathComponent getComponent() { result = component }
+
+  override DataFlowType getTypeImpl() { result = getDataFlowType(component.getComponentType()) }
```

Comment:
Do we really need to implement `getTypeImpl` on all the post-update nodes? Can the result of `getTypeImpl` on a post-update node not simply be the result of calling `getTypeImpl` on the pre-update node?

Reply:
Oh, I see what we do actually implement this predicate on `PostUpdateNode`s: https://github.com/github/codeql/pull/14592/files#diff-514dd0988f7b47e40ff8c1e66ea886d1f947d047da108270ed0ddec75c2856c1R1531. Do we really need this (and similar) overrides then?

Reply:
That was actually only on `ExprPostUpdateNode`s. Moving it to `PostUpdateNodeImpl` simplifies some things, although it leads to a couple of duplicate overrides. It also fixed a bug with the postupdate nodes on keypaths.